src/components/use/column-options.js

```diff
@@ -0,0 +1,53 @@
+import useModel from './model';
+import useLayout from './layout';
+
+const genericObjectProperties = {
+  qInfo: {
+    qId: 'MeasureList',
+    qType: 'MeasureList',
+  },
+  qMeasureListDef: {
+    qType: 'measure',
+    qData: {
+      title: '/qMetaDef/title',
+      expression: '/qMeasure/qDef',
+    },
+  },
+  qFieldListDef: {
+    qShowDerivedFields: true,
+  },
+  qDimensionListDef: {
+    qType: 'dimension',
+    qData: {
+      title: '/qMetaDef/title',
+      all: '/qDim',
+      tags: '/qMetaDef/tags',
+      grouping: '/qDim/qGrouping',
+      info: '/qDimInfos',
+      labelExpression: '/qDim/qLabelExpression',
+    },
+  },
+};
+
+export function layoutToList(layout) {
+  return layout;
+}
+
+export default function useColumnOptions(app) {
+  const model = useModel(app, genericObjectProperties);
+  const layout = useLayout(model);
+  if (layout) {
+    const measures = layout.qMeasureList.qItems ? layout.qMeasureList.qItems.map(measure => ({ title: measure.qData.title, type: 'measure', hyperCubeContent: { qLibraryId: measure.qInfo.qId } })) : [];
```

Comment:
You should probably use a state for these calculations (i.e. do them inside `useEffect()` conditionally).